go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -144,6 +147,12 @@ func (st *symtab) AddVindexTable(alias sqlparser.TableName, vindexTable *vindexe
 // At this point, only tables and uniqueColumns are set.
 // All other fields are ignored.
 func (st *symtab) Merge(newsyms *symtab) error {
+	if st.orderedTables == nil || newsyms.orderedTables == nil {
+		// This code is unreachable because there are higher level
+		// protections that disallow intermixing of symtabs with
+		// no tables with symtabs that have tables.
```

Comment:
If this is truly unreachable why not return an appropriate error?

Reply:
Actually, the code is reachable for joining an `information_schema` table with another. I've updated the comment. Nevertheless, the case itself is not an error condition. The behavior is as intended. I somehow assumed the higher level checks made this code path unreachable.